src/use_case/settings/SettingsInputData.java

```diff
@@ -1,16 +1,23 @@
 package use_case.settings;
 
-import entities.auth.AbstractUser;
+import entities.auth.User;
 
 public class SettingsInputData {
 
-  AbstractUser user;
+  User user;
 
-  public SettingsInputData(AbstractUser user) {
+  String updatedUsername;
+
+  public SettingsInputData(String updatedUsername, User user) {
+    this.updatedUsername = updatedUsername;
     this.user = user;
   }
 
-  AbstractUser getUser() {
+  String getUpdatedUsername() {
+    return updatedUsername;
```

Comment:
Wrong tense. You don't need to fix this, but it should be something like `newUsername`. You could use `updatedUsername` in the output data

Reply:
Fixed